### PR TITLE
Logout

### LINKED ISSUE #159: 로그아웃
- nav에 로그아웃 버튼 만들기
- 로그아웃 후 로그인 페이지로 이동

### PR DESCRIPTION
**Pull request by gyim1345**
1. Add Logout feature

Add logout button and its features
- set loggedIn to false on click
- redirect to login page if not loggedin
- remove token from local storage on click

2. Add check login
    
check login on every App rerender
check if token exists in local storage
set loggedIn to true if token exists
redirect to /issue/list if loggedIn
redirect to login page if no loggedin

3. Fix login Page authorization 

Fix error where on page entry, it requests to login w/o clicking login button

### CODE REVIEW

**Review by profornnan — APPROVED** [review]
고생하셨습니다!!

**Review by pieisland — APPROVED** [review]
logout!

File: front/src/pages/LoginPage.js [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 import GitHubSvg from '../svgs/GitHubSvg';
 import SearchSvg from '../svgs/SearchSvg';
 import PersonSvg from '../svgs/PersonSvg';
+import queryString from 'query-string'
+import Auth from '../apis/Auth.api';
 
-const StyledLink = styled(Link)`
+const GITHUB_AUTHORIZATION_URL = `https://github.com/login/oauth/authorize?client_id=${process.env.GITHUB_CLIENT_ID}&redirect_uri=${process.env.HOMEPAGE_URL}`
+
+const StyledLink = styled.a`
```
Comment (rockpell) [inline front/src/pages/LoginPage.js:12]: Link에서 a 태그로 바뀌면서 StyledLink라는 이름이 헷갈릴수도 있을거 같네요

**Review by rockpell — APPROVED** [review]
수고하셨습니다!

**Pull request merged**